src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOnSession.java

```diff
@@ -160,10 +161,14 @@ public void put(SecurityIdentity identity) {
                             connection.getInputStream().close();
                         } catch (Exception cause) {
                             log.warnHttpMechSsoFailedLogoutParticipant(remoteURI.toString(), cause);
+                            // failed to logout participant, remove it from the list of participants
+                            try (SingleSignOn target = context.getSingleSignOnManager().find(id)) {
+                                target.removeParticipant(participantId);
+                            }
                         }
                     });
 
-                    try (SingleSignOn target = this.context.getSingleSignOnManagerManager().find(id)) {
+                    try (SingleSignOn target = this.context.getSingleSignOnManager().find(id)) {
                         if (target != null) {
                             // If all logout requests were successful, then there should be no participants, and we can invalidate the SSO
                             if (target.getParticipants().isEmpty()) {
```

Comment:
Rather than perform a separate lookup/close on failure (above), it would be better to collect the failed participant ids and just remove them in this block, i.e. prior to invalidation.  This isEmpty() check becomes redundant.